enki/PhysicalEngine.h

```diff
@@ -393,6 +396,8 @@ namespace Enki
 		virtual void doGlobalInteractions(double dt, World* w);
 		//! Sort local interactions. Called by addLocalInteraction ; can be called by subclasses in case of interaction radius change.
 		void sortLocalInteractions(void);
+
+		virtual void clickedInteraction(bool pressed, int buttonCode, double pointX, double pointY, double pointZ){};
```

Comment:
Is there a reason why only robots have this method?

Also, `buttonCode` should be a bit field, and an `unsigned`.